internal/kafka/internal/handlers/admin_kafka.go

```diff
@@ -138,18 +141,89 @@ func (h adminKafkaHandler) Update(w http.ResponseWriter, r *http.Request) {
+
+				if ek != nil {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to compare desired kafka version: %s with actual kafka version: %s", kafkaRequest.DesiredKafkaVersion, currentKafkaVersion))
+				}
+
+				// no minor/ major version downgrades allowed for kafka version
+				if vCompKafka > 0 {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to downgrade kafka: %s version: %s to the following kafka version: %s", kafkaRequest.ID, currentKafkaVersion, kafkaRequest.DesiredKafkaVersion))
+				}
+
 				return nil
 			},
 			func() *errors.ServiceError { // Validate DesiredStrimziVersion
 				if kafkaRequest.DesiredStrimziVersion != kafkaUpdateReq.StrimziVersion && kafkaUpdateReq.StrimziVersion != "" && kafkaRequest.StrimziUpgrading {
 					return errors.New(errors.ErrorValidation, "Unable to update strimzi version. Another upgrade is already in progress.")
 				}
+
 				return nil
 			},
 			func() *errors.ServiceError { // Validate DesiredKafkaIBPVersion
 				if kafkaRequest.DesiredKafkaIBPVersion != kafkaUpdateReq.KafkaIbpVersion && kafkaUpdateReq.KafkaIbpVersion != "" && kafkaRequest.KafkaIBPUpgrading {
 					return errors.New(errors.ErrorValidation, "Unable to update ibp version. Another upgrade is already in progress.")
 				}
+
+				currentIBPVersion, _ := arrays.FirstNonEmpty(kafkaRequest.ActualKafkaIBPVersion, kafkaRequest.DesiredKafkaIBPVersion)
+				vCompOldNewIbp, eIbp := api.CompareBuildAwareSemanticVersions(currentIBPVersion, kafkaRequest.DesiredKafkaIBPVersion)
+
+				if eIbp != nil {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to compare actual ibp version: %s with desired ibp version: %s", currentIBPVersion, kafkaRequest.DesiredKafkaVersion))
+				}
+
+				// actual ibp version cannot be greater than desired ibp version (no downgrade allowed)
+				if vCompOldNewIbp > 0 {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to downgrade kafka: %s ibp version: %s to a lower version: %s", kafkaRequest.ID, kafkaRequest.DesiredKafkaIBPVersion, currentIBPVersion))
+				}
+
+				vCompIbpKafka, eIbpK := api.CompareBuildAwareSemanticVersions(kafkaRequest.DesiredKafkaIBPVersion, kafkaRequest.DesiredKafkaVersion)
+
+				if eIbpK != nil {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to compare kafka ibp version: %s with kafka version: %s", kafkaRequest.DesiredKafkaIBPVersion, kafkaRequest.DesiredKafkaVersion))
+				}
+
+				// ibp version cannot be greater than kafka version
+				if vCompIbpKafka > 0 {
+					return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to update kafka: %s ibp version: %s with kafka version: %s", kafkaRequest.ID, kafkaRequest.DesiredKafkaIBPVersion, kafkaRequest.DesiredKafkaVersion))
+				}
+
+				return nil
+			},
+			func() *errors.ServiceError { // Validate Strimzi Version
+				cluster, err := h.clusterService.FindClusterByID(id)
```

Comment:
This will need to be changed to
 ```diff
- cluster, err := h.clusterService.FindClusterByID(id)
+ cluster, err := h.clusterService.FindClusterByID(kafkaRequest.ID) 
```

Reply:
```suggestion
				cluster, err := h.clusterService.FindClusterByID(kafkaRequest.ClusterID)
```

I think it'll have to be something like above

Reply:
Thanks for confirming. The change will be needed. Is it okay to push the change? 
We'll need to investigate further on why the integration tests are failing and where exactly they are failing.